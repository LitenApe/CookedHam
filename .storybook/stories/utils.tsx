import React, { ReactElement } from "react";

const horizontalCenter = {
  display: "flex",
  justifyContent: "center",
};

export const CenterContent = (storyFn: () => JSX.Element): ReactElement => (
  <div style={horizontalCenter}>{storyFn()}</div>
);
