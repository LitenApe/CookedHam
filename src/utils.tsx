let lastId = 0;

export function newId(prefix = "id"): string {
  lastId += 1;
  return `${prefix}${lastId}`;
}
