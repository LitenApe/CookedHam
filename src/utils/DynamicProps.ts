import { ComponentProps } from 'react';

/**
 * Redeclare forwardRef to better support generic types
 */
declare module 'react' {
  function forwardRef<T, P = {}>(
    render: (props: P, ref: React.Ref<T>) => React.ReactElement | null
  ): (props: P & React.RefAttributes<T>) => React.ReactElement | null;
}

export type HTMLTags = keyof JSX.IntrinsicElements;

export type DynamicProps<T extends HTMLTags> = {
  as?: T;
} & ComponentProps<T>;
