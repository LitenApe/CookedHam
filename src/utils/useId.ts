import { useMemo } from 'react';

const _used: { [key: string]: number } = {};

function useId(prefix: string = 'kitchen') {
  const id = useMemo(() => {
    if (_used[prefix] === undefined) {
      _used[prefix] = 1;
      return `${prefix}-1`;
    } else {
      _used[prefix] += 1;
      return `${prefix}-${_used[prefix]}`;
    }
  }, [prefix]);

  return id;
}

export default useId;
