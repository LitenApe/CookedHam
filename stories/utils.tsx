import React from 'react';
import { RenderFunction } from '@storybook/react';

const horizontalCenter = {
  display: 'flex',
  justifyContent: 'center',
}

export const CenterContent = (storyFn: RenderFunction) => (
  <div style={horizontalCenter}>{storyFn()}</div>
);
