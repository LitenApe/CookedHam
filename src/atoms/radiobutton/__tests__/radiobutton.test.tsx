import React from "react";
import TestRenderer from "react-test-renderer";
import { shallow, } from "enzyme";
import { RadioButton, } from "../index";
import * as util from "../../../utils";

jest.mock("../../../utils");

describe("RadioButton", () => {
  beforeEach(() => {
    const mock = jest.spyOn(util, "newId");
    mock.mockReturnValue("radio-1");
  });

  it("renders without crashing", () => {
    TestRenderer.create(<RadioButton label="label" />);
  });

  it("renders with label", () => {
    const radiobutton = TestRenderer.create(<RadioButton label="label" />);
    expect(radiobutton).toMatchSnapshot();
  });

  it("is checked when 'checked' prop is passed", () => {
    const radiobutton = TestRenderer.create(<RadioButton label="label" checked />);
    expect(radiobutton).toMatchSnapshot();
  });

  it("is disabled when 'disabled' prop is passed", () => {
    const radiobutton = TestRenderer.create(<RadioButton label="label" disabled />);
    expect(radiobutton).toMatchSnapshot();
  });

  it("sends change event on change", () => {
    const mockFn = jest.fn();
    const radiobutton = shallow(<RadioButton label="label" onChange={mockFn} />);
    radiobutton.find("input").simulate("change");
    expect(mockFn).toBeCalled();
  });

  it("sends radio value on change", () => {
    const mockFn = jest.fn();
    const radiobutton = shallow(<RadioButton label="label" value="value" onChange={mockFn} />);
    radiobutton.find("input").simulate("change");
    expect(mockFn.mock.calls[0][1]).toBe("value");
  });

  it("sends 'true' on change when not checked", () => {
    const mockFn = jest.fn();
    const radiobutton = shallow(<RadioButton label="label" value="value" onChange={mockFn} />);
    radiobutton.find("input").simulate("change");
    expect(mockFn.mock.calls[0][2]).toBeTruthy();
  });

  it("sends 'false' on change when checked", () => {
    const mockFn = jest.fn();
    const radiobutton = shallow(<RadioButton label="label" value="value" onChange={mockFn} checked />);
    radiobutton.find("input").simulate("change");
    expect(mockFn.mock.calls[0][2]).toBeFalsy();
  });
});
