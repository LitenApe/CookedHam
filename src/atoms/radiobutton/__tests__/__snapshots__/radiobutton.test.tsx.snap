// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton is checked when 'checked' prop is passed 1`] = `
<label
  className="ch-radio-button checked"
  htmlFor="radio-1"
  onKeyPress={[Function]}
  tabIndex={0}
>
  <input
    checked={true}
    id="radio-1"
    onChange={[Function]}
    type="radio"
  />
  label
</label>
`;

exports[`RadioButton is disabled when 'disabled' prop is passed 1`] = `
<label
  className="ch-radio-button disabled"
  htmlFor="radio-1"
  onKeyPress={[Function]}
  tabIndex={-1}
>
  <input
    disabled={true}
    id="radio-1"
    onChange={[Function]}
    type="radio"
  />
  label
</label>
`;

exports[`RadioButton renders with label 1`] = `
<label
  className="ch-radio-button"
  htmlFor="radio-1"
  onKeyPress={[Function]}
  tabIndex={0}
>
  <input
    id="radio-1"
    onChange={[Function]}
    type="radio"
  />
  label
</label>
`;
